src/modules/dashboard/components/TrendingChannelItemList.tsx

```diff
@@ -0,0 +1,52 @@
+// React and other libraries
+import { useEffect, useState } from 'react';
+
+//Hooks
+import { useGetTrendingChannels } from 'queries/hooks';
+
+//Utility functions
+import { getTrendingChannelsData } from '../Dashboard.utils';
+
+// Components
+import { Box, Separator } from 'blocks';
+import { ChannelItem } from './ChannelItem';
+import { TrendingSource, TrendingStartTime } from '../Dashboard.constants';
+import { appConfig } from 'config';
+import { EnvKeys, SourceKeys } from '../Dashboard.types';
+
+const TrendingChannelItemList = () => {
+  const [trendingChannels, setTrendingChannels] = useState<Array<string>>([]);
+
+  const startDate = new Date(TrendingStartTime[appConfig.appEnv as EnvKeys]);
+  const firstEndDate = new Date(Date.now()).toISOString().split('T')[0];
+  const secondEndDate = new Date(Date.now() - 7 * 86400000).toISOString().split('T')[0];
+  const { data: currentData, isLoading } = useGetTrendingChannels({
+    startDate,
+    endDate: firstEndDate,
+    channel: 'All',
+    source: TrendingSource[appConfig.appEnv as EnvKeys],
+  });
+  const { data: weekData } = useGetTrendingChannels({
+    startDate,
+    endDate: secondEndDate,
+    channel: 'All',
+    source: TrendingSource[appConfig.appEnv as EnvKeys],
+  });
+
+  useEffect(() => {
+    const channelsData = getTrendingChannelsData(weekData, currentData);
+    setTrendingChannels(channelsData);
+  }, [currentData, weekData]);
+  return trendingChannels?.map((channel, index) => (
```

Comment:
@mishramonalisha76 I don't understand this? Why we doing this twice?  Can't this logic of modelling the data be moved to mode creators?

If this is required please add comments in codebase.